src/mlpack/core/data/load_impl.hpp

```diff
@@ -63,8 +63,8 @@ void TransposeTokens(std::vector<std::vector<std::string>> const &input,
 
 } // namespace details
 
-template<typename eT>
-bool inline inplace_transpose(arma::Mat<eT>& X)
+template <typename eT, template <typename> typename MatType>
+bool inline inplace_transpose(MatType<eT>& X)
```

Comment:
Why not just remove the `eT` parameter and use `MatType` as the only template parameter?